client/src/data/types/request.ts

```diff
@@ -11,4 +12,5 @@ export default interface Request {
     fulfilledAt?: Date;
     deleted?: boolean;
     fulfilled?: boolean;
+    matchedDonations?: DonationFormContributionTuple[];
```

Comment:
We would need new resolvers for matching right?

Resolver would:

- update quantityRemaining on DonationForm
- update/add to matchedDonations array

Anything I'm missing? I can start working on this

Reply:
I'm not entirely sure if we need a separate resolver for matching - what I'm thinking is that the client-side would fetch the donation form and requests, and keep a copy of the originals. 

When the user interacts with the matching page, we'll modify one of the copies then when the user clicks "Confirm matches", we can use the `updateDonationForm` and `updateRequest` mutations with the new matchedDonations array and new quantityRemaining.

Keeping a copy of the original requests allows us to diff the new requests and originals to see which ones have changed (i.e. which requests to call `updateRequest` on).

lmk if that makes sense!

Reply:
Sounds good!